java/code/src/com/suse/manager/webui/Router.java

```diff
@@ -117,204 +106,49 @@ public void init() {
         AdminApiController.initRoutes();
 
         // Minions
-        get("/manager/systems/keys",
-                withUserPreferences(withCsrfToken(withUser(MinionController::list))),
-                jade);
-        get("/manager/systems/bootstrap",
-                withCsrfToken(withOrgAdmin(MinionController::bootstrap)),
-                jade);
-        get("/manager/systems/cmd",
-                withCsrfToken(MinionController::cmd),
-                jade);
-        get("/manager/systems/:id",
-                MinionController::show);
+        MinionController.initRoutes(jade);
 
         // Minions API
-        post("/manager/api/systems/bootstrap", withOrgAdmin(MinionsAPI::bootstrap));
-        post("/manager/api/systems/bootstrap-ssh", withOrgAdmin(MinionsAPI::bootstrapSSH));
-        get("/manager/api/systems/keys", withUser(MinionsAPI::listKeys));
-        post("/manager/api/systems/keys/:target/accept", withOrgAdmin(MinionsAPI::accept));
-        post("/manager/api/systems/keys/:target/reject", withOrgAdmin(MinionsAPI::reject));
-        post("/manager/api/systems/keys/:target/delete", withOrgAdmin(MinionsAPI::delete));
-
-        // States
-        get("/manager/systems/details/packages",
-                withCsrfToken(MinionController::packageStates),
-                jade);
-        get("/manager/systems/details/custom",
-                withCsrfToken(MinionController::minionCustomStates),
-                jade);
-        get("/manager/systems/details/highstate",
-                withCsrfToken(withUser(MinionController::highstate)),
-                jade);
-        get("/manager/multiorg/details/custom",
-                withCsrfToken(MinionController::orgCustomStates),
-                jade);
-        get("/manager/yourorg/custom",
-                withCsrfToken(withUser(MinionController::yourOrgConfigChannels)),
-                jade);
-        get("/manager/groups/details/custom",
-                withCsrfToken(withUser(MinionController::serverGroupConfigChannels)),
-                jade);
-        get("/manager/groups/details/highstate",
-                withCsrfToken(withUser(MinionController::serverGroupHighstate)), jade);
-
-        // SSM API
-        get("/manager/systems/ssm/highstate",
-                withCsrfToken(withUser(MinionController::ssmHighstate)), jade);
-        get("/manager/systems/ssm/channels/bases",
-                withUser(SsmController::getBaseChannels));
-        post("/manager/systems/ssm/channels/allowed-changes",
-                withUser(SsmController::computeAllowedChannelChanges));
-        post("/manager/systems/ssm/channels",
-                withUser(SsmController::changeChannels));
+        MinionsAPI.initRoutes();
```

Comment:
How about organizing the removed code by route i.e. `groups`, `systems`, etc by adding new methods and calling them inside `MinionsAPI.initRoutes()` or calling them directly from here.

Reply:
Split that one into private functions